.gitlab/e2e.yml

```diff
@@ -2,11 +2,18 @@
 # e2e stage
 # Contains jobs which runs e2e tests on our Docker images.
 
+.container_needs:
```

Comment:
I'd make this more explicit, with a name like `.container_e2e_needs` or `.e2e_container_needs`. The reason being that this block _could_ be mistakenly used elsewhere.